aks-node-controller/parser/helper.go

```diff
@@ -613,6 +609,68 @@ func getServicePrincipalFileContent(authConfig *aksnodeconfigv1.AuthConfig) stri
 	return base64.StdEncoding.EncodeToString([]byte(authConfig.GetServicePrincipalSecret()))
 }
 
+func getKubeletFlags(kubeletConfig *aksnodeconfigv1.KubeletConfig) string {
+	return createSortedKeyValuePairs(kubeletConfig.GetKubeletFlags(), " ")
+}
+
+// pass the type of v as an input arguem
+func marshalToJson(v any) ([]byte, error) {
+	// Originally we can set the Multiline here and it will marshal to a JSON we can use.
+	// However the protojson team intentionally randomly add extra whitespace after the key in the key-value.
+	// E.g., Sometimes it is "key": "value" and sometimes it is "key":  "value".
+	// They did it intentionally to make the output formot not reliable,
+	// because they think it is not a good idea to rely on the JSON output format.
+	// ref: https://github.com/protocolbuffers/protobuf-go/commit/582ab3de426ef0758666e018b422dd20390f7f26
+	marshaler := &protojson.MarshalOptions{
+		Indent: "",
+	}
+
+	// Therefore, we have to implement our own re-formatting to make the output reliable as we expected.
+	switch v := v.(type) {
+	case *aksnodeconfigv1.KubeletConfigFileConfig:
+		data, err := marshaler.Marshal(v)
+		if err != nil {
+			log.Printf("error marshalling: %v", err)
+			return nil, err
+		}
+
+		var rawMessage json.RawMessage = data
+		jsonByte, err := json.MarshalIndent(rawMessage, "", "  ")
+		if err != nil {
+			log.Printf("error marshalling kubelet config file content: %v", err)
+			return nil, err
+		}
+		jsonByte = append(jsonByte, '\n')
+		return jsonByte, nil
+	}
+	return nil, fmt.Errorf("unsupported type: %T", v)
```

Comment:
nit: use default case for this?

Reply:
I tried to set a default case. But then the protojson.Marshal() will complain because it strictly needs a type that is defined from protobuf so that it can handle the json key naming correctly. If in the future, more protobuf defined types want to use this function, we will need to add it explicitly with 1 more line/case.